src/handlers/pr_tracking.rs

```diff
@@ -210,30 +222,57 @@ pub async fn get_assigned_prs(ctx: &Context, user_id: UserId) -> HashSet<PullReq
 
 /// Add a PR to the workqueue of a team member.
 /// Ensures no accidental PR duplicates.
-async fn upsert_pr_into_workqueue(ctx: &Context, user_id: UserId, pr: PullRequestNumber) {
-    ctx.workqueue
-        .write()
-        .await
-        .reviewers
-        .entry(user_id)
-        .or_default()
-        .insert(pr);
+///
+/// Returns true if the PR was actually inserted.
+fn upsert_pr_into_user_queue(
+    workqueue: &mut RwLockWriteGuard<ReviewerWorkqueue>,
+    user_id: UserId,
+    pr: PullRequestNumber,
+) -> bool {
+    workqueue.reviewers.entry(user_id).or_default().insert(pr)
 }
 
-/// Delete a PR from the workqueue of a team member
-async fn delete_pr_from_workqueue(ctx: &Context, user_id: UserId, pr: PullRequestNumber) {
-    let mut queue = ctx.workqueue.write().await;
-    if let Some(reviewer) = queue.reviewers.get_mut(&user_id) {
-        reviewer.remove(&pr);
+/// Delete a PR from the workqueue of a team member.
+fn delete_pr_from_user_queue(
+    workqueue: &mut ReviewerWorkqueue,
+    user_id: UserId,
+    pr: PullRequestNumber,
+) {
+    if let Some(queue) = workqueue.reviewers.get_mut(&user_id) {
+        queue.remove(&pr);
     }
 }
 
+/// Delete a PR from the workqueue completely.
+fn delete_pr_from_all_queues(workqueue: &mut ReviewerWorkqueue, pr: PullRequestNumber) {
+    for queue in workqueue.reviewers.values_mut() {
+        queue.retain(|pr_number| *pr_number != pr);
+    }
+}
+
+/// Returns true if the workqueue should assume that this PR is actually waiting for a reviewer.
+/// The function receives atomic attributes so that it is compatible both with triagebot's
+/// `Issue` struct (used for incremental updates) and octocrab's `PullRequest` struct (used for
+/// batch PR loads).
+///
+/// Note: this functionality is currently hardcoded for rust-lang/rust, other repos might use
+/// different labels.
+fn waits_for_a_review(labels: &[Label], is_open: bool, is_draft: bool) -> bool {
+    let is_blocked = labels
+        .iter()
+        .any(|l| l.name == "S-blocked" || l.name == "S-inactive");
+    let is_rollup = labels.iter().any(|l| l.name == "rollup");
+    let is_waiting_for_reviewer = labels.iter().any(|l| l.name == "S-waiting-on-review");
```

Comment:
I would prefer if this logic was not hard-coded in the code (yes, I know all this is pretty r-l/r specific anyway but still). Can we at least have "positive" labels (`S-waiting-on-review`) and "negative" labels in a `const` slice? Or better, make them configurable in `ReviewPrefsConfig`?

Reply:
Makes me sad that we have to hard-code them, but oh well.